src/mapgen_v6.cpp

```diff
@@ -73,7 +74,19 @@ MapgenV6::MapgenV6(int mapgenid, MapgenParams *params, EmergeManager *emerge)
 	noise_height_select  = new Noise(&sp->np_height_select,  seed, csize.X, csize.Y);
 	noise_mud            = new Noise(&sp->np_mud,            seed, csize.X, csize.Y);
 	noise_beach          = new Noise(&sp->np_beach,          seed, csize.X, csize.Y);
-	noise_biome          = new Noise(&sp->np_biome,          seed, csize.X, csize.Y);
+
+	noise_biome    = new Noise(
+		&sp->np_biome,
+		seed,
+		csize.X + 2 * MAP_BLOCKSIZE,
+		csize.Y + 2 * MAP_BLOCKSIZE
+	);
+	noise_humidity = new Noise(
+		&sp->np_humidity,
+		seed,
+		csize.X + 2 * MAP_BLOCKSIZE,
+		csize.Y + 2 * MAP_BLOCKSIZE
+	);
```

Comment:
This is how function calls taking table arguments are done in Lua, but these should still be called the same way as the other C++ function calls), with the closing ); on the same line as the last argument.